src/serverMain/kotlin/com/studo/campusqr/endpoints/ReportData.kt

```diff
@@ -95,4 +93,38 @@ suspend fun AuthenticatedApplicationCall.returnReportData() {
           impactedUsersEmailsCsvFileName = "${csvFilePrefix?.plus("-emails") ?: "emails"}.csv"
       )
   )
+}
+
+suspend fun AuthenticatedApplicationCall.listAllActiveCheckIns() {
+  if (!user.isModerator) {
+    respondForbidden()
+    return
+  }
+
+  val params = receiveJsonStringMap()
+  val emailAddress = params.getValue("emailAddress")
+
+  val checkIns = runOnDb {
+    val nextInfectionHours: Int = getConfig("nextInfectionHours")
+    getCollection<CheckIn>()
+        .find(CheckIn::email equal emailAddress, CheckIn::date greater Date().addHours(-nextInfectionHours))
+        .sortByDescending(CheckIn::date)
+        .toList()
+  }
+
+  val locationMap = runOnDb {
```

Comment:
should we do a shortcut if the user has no active check-ins? i guess this would happen a lot?

Reply:
early optimization